### PR TITLE
PO / SD upload rework

### PR DESCRIPTION
**Pull request by Lubosky**
## 📝 A short description of the changes
 - added ability for admins/assessors to add/remove docs to and from shortlisted applications
 -  fixed audit log when doc is removed
 - added completion status (can be changed for PO / SD applications)

## 🔗 Link to the relevant story (or stories)
Asana card: https://app.asana.com/0/1200504523179345/1203129837638856/f

## :shipit: Deployment implications
None

## ✅ Checklist

- [x] Features that cannot go live are behind a feature flag/env var or specify deploy date and open PR as draft 
- [x] I have checked that commit messages make sense and explain the reasoning for each change
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have squashed any unnecessary or part-finished commits

### CODE REVIEW

File: app/assets/javascripts/admin/form_answers.js.coffee [hunk 1]
```diff
@@ -25,13 +25,36 @@ ready = ->
...
+
     wrapper = $("#audit-certificate-form")
     $(".attachment-link", wrapper).removeClass("if-js-hide")
     $(".attachment-link", wrapper).addClass("btn btn-default btn-block btn-attachment")
     $(".attachment-link", wrapper).prepend("<span class='btn-title'>Attach External Accountant's Report</span>")
     $(".attachment-link", wrapper).prepend("<span class='glyphicon glyphicon-paperclip'></span>")
     $(".attachment-link", wrapper).prependTo("#new_audit_certificate")
 
+  toggleCommercialFiguresButtonVisibility = ->
+    console.log($('.commercial-figures-file').length == 0)
```
Comment (TheDancingClown) [inline app/assets/javascripts/admin/form_answers.js.coffee:52]: ✂️

File: app/views/admin/figures_and_vat_returns/_form.html.slim [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+.span
+  = hidden_field_tag "valid", "false", id: "#{attachment.model_name.param_key}-valid"
+= form_for [namespace_name, form_answer, attachment], html: { multipart: true } do |form|
+  .well.js-attachment-form
+    h3 Attach document
+
+    - if form.object.errors.any?
+      .errors = form.object.errors.full_messages.join(", ")
+
+    ul.list-unstyled
```
Comment (TheDancingClown) [inline app/views/admin/figures_and_vat_returns/_form.html.slim:13]: Why is there a <ul> tag here with no nested list?

**Review by TheDancingClown — CHANGES_REQUESTED** [review]

File: app/assets/javascripts/admin/form_answers.js.coffee (hunk 1 above)
Comment (Lubosky) [reply app/assets/javascripts/admin/form_answers.js.coffee]: Done. Thanks for spotting it.

**Review by TheDancingClown — APPROVED** [review]

File: app/views/admin/figures_and_vat_returns/_form.html.slim (hunk 2 above)
Comment (Lubosky) [reply app/views/admin/figures_and_vat_returns/_form.html.slim]: Not sure, tried to look if there is some code which would attach something via JS but nothing. So removed it from all places as it seems to be redundant.

**Pull request merged**